include/verilated_vpi.cpp

```diff
@@ -224,7 +224,13 @@ class VerilatedVpioVar : public VerilatedVpio {
     vluint32_t entSize() const { return m_entSize; }
     vluint32_t index() { return m_index; }
     virtual vluint32_t type() const {
-        return (varp()->dims() > 1) ? vpiMemory : vpiReg;  // but might be wire, logic
+        if (varp()->isPublicRD()) {
```

Comment:
This probably needs to change to something else depending on AstVarType GPARAM, LPARAM.

Reply:
Yes, I agree. We should check for the actual type, otherwise this just leads to another issue once someone used public rd. This info probably needs to be emitted and constructed at simulation time.